### ISSUE TITLE
Stream 'Games' instead of building them with the App

### ISSUE THREAD

**Issue by AlexSolbergRam** [Ramboll/ppm_rdk-gamificationplatform-sdk_unity#29]

**Comment by eldamir** [comment]
@marcusdyrholm, I didn't hear back from you regarding the format of the URLs you wanted

**Comment by eldamir** [comment]
Note: We can in fact add our API token on the addressables HTTP request by modifying the web request in the callback designed for it: https://docs.unity3d.com/Packages/com.unity.addressables@1.20/manual/TransformInternalId.html#webrequest-override

**Comment by eldamir** [comment]
Note: The Url format seems to be `http://staging.ramrisk.dk:8080/[BuildTarget]`. So either:

- `http://staging.ramrisk.dk:8080/IOS`
- `http://staging.ramrisk.dk:8080/Android`

And later, possibly:

- `http://staging.ramrisk.dk:8080/WebGl`

**Comment by eldamir** [comment]
@marcusdyrholm. this is a bit problematice, actually. Very easy to make mistakes

> Update a Build
> 
> 2. 1. Check Restrictions
> It is very important that when you update a build that you use the Addressables Toolkit. If you just build regularly you will reset the adressables catalog load path and the name of the catalog. The file name of the catalog is built with the app. Firstly check if there is any update restrictions by clicking on Tools in the Addressables Groups window and select "Check for Content Update Restrictions". Then select the build target folder in the file explorer window and then the .bin file associated with your build target. In the window that pops up try to fix any issues if there are any.
> 
> 2. Update Bundle
> In the Addressables Group window click the Build button and select Update a Previous Build. Select the .bin file and wait for the update to finish. Updating a build will edit your catalog file without changing the name of the catalog file. The catalog file name is built into the app when it is built to the device and cannot be changed at runtime. Updating a build will create a new asset bundle file.
> 
> The catalog file must always have the same name as the catalog file that was built with the last build of the app.

As I understand, in order to properly re-build or "update a build", you need to have some version of a previous build, so you can provide the .bin file? So for me, working on this locally right now. I did not have a previous build, so I just clicked "New build". As soon a some build is hosted on prod, we need to make sure that the app is built and distributed with knowledge of that specific catalog name, and for all future builds, we need to do this careful updating, using the old .bin file to rebuild without resetting catalog names.

So in short, we need to all have some version of the .bin file for the catalog, once we go to prod. This is a file that is built locally, and should probably not be checked in.

That is how I understood it, at least... Although, now looking at the changed files after doing a "New build", I see that the mentioned .bin file is not in the "ServerData/Android" folder (part of the bundle), but rather in the addressable state folder. So I guess we already have the .bin under source control, so we all can se it, and Plastic just told me I updated, so maybe "New build" was already a mistake, and I should have done the careful update instead.

This has largely been me talking to myself, but feel free to chime in 🤣

**Comment by eldamir** [comment]
Tried the careful update but got this

![Image](https://user-images.githubusercontent.com/2298514/193975504-4c44e1f3-23f7-401f-b0b6-1304a5a9928a.png)

I'll just do my testing with a fresh build. Should work fine locally... But we probably need to clear up any confusion here, so we don't screw up in production later

**Comment by eldamir** [comment]
Note: A lot of good info about managing remote content here: https://docs.unity3d.com/Packages/com.unity.addressables@1.19/manual/RemoteContentDistribution.html

**Comment by eldamir** [comment]
@marcusdyrholm and @AlexSolbergRam. The admin portal part of all of this is not implemented and deployed



![Image](https://user-images.githubusercontent.com/2298514/194050628-3985364b-0f2f-423a-ac5e-6e0559367729.png)


The small configuration changes needed for the app to talk to the backend is implemented here:



They contain:

- A little fix to Marcus' fetcher, so that the API token is included, so we can authorize access
- A "Production" profile to the addressables that know where to fetch addressables.


This, incidentally, is also an important first step towards managing translations in a Unity-native format, or at the very least, to have more control, so we can pull translations away from Google Sheets. As previously mentioned ad-nauseum: We are not allowed to hold Ramboll data on Google's services, and additionally, it has also gotten Jesper's attention, and he wants us to get stuff over at POeditor as soon as feasible
